commands/time.rb

```diff
@@ -3,28 +3,17 @@
 msg_timezone_not_set = "%<nick>s has not set his or her timezone"
 msg_timezone_not_found = "Could not find timezone %<timezone>s"
 
-with_nick = proc do |&block|
-  proc do
-    nickname = params[:nick].gsub(/\:me/i, sender.nick)
-    if nick = Scarlet::Nick.first(nick: nickname)
-      instance_exec(nick, &block)
-    else
-      reply "Cannot find Nick #{nickname}"
-    end
-  end
-end
-
 find_timezone = proc do |timezone|
   Time.find_zone timezone
 end
 
 hear (/set(?:\s+my)?\s+timezone\s+(?<timezone>.+)/i) do
-  clearance :registered
+  clearance &:registered?
   description 'Sets your timezone. (used with "time for" command)'
   usage 'set [my] timezone <timezone>'
   on do
     timezone = params[:timezone]
-    if nick = Scarlet::Nick.first(nick: sender.nick)
+    with_nick sender.nick do |nick|
```

Comment:
This looks confusing. I'm thinking about somehow merging the distinction between the channel user and `Scarlet::Nick` (because Nick is actually a registered user right now). So sort of like having a registered flag on the instances, and only persisting those to the DB. WDYT? Might be overkill.